docs/api/csf/csf-factories.mdx

```diff
@@ -0,0 +1,305 @@
++ });
+- };
+- export default config;
+```
+
+### 3. Update your preview config file
+
+Update your `.storybook/preview.ts` file to use the new [`definePreview`](#definepreview) function.
+
+```diff title=".storybook/preview.ts"
+// Replace your-framework with the framework you are using (e.g., react-vite, nextjs, experimental-nextjs-vite)
++ import { definePreview } from '@storybook/your-framework/node';
+- import { Preview } from '@storybook/your-renderer';
+// 👇 Import the addons you are using
++ import addonTest from '@storybook/experimental-addon-test';
+
++ export default definePreview({
+- export const preview: Preview = {
+    // ...current config
+    // 👇 Add your addons here
++   addons: [addonTest()],
++ });
+- };
+- export default preview;
+```
+
+### 4. Update your story files
+
+Story files have been updated for improved usability. With the new format:
+
+- Import the preview construct from the Storybook preview file
+- The meta object is now created via the [`preview.meta`](#previewmeta) function and does not have to be exported as a default export
+- Stories are now created from the meta object, via the [`meta.story`](#metastory) function
+
+```diff title="Button.stories.ts"
+// Learn about the # subpath import: https://storybook.js.org/docs/api/csf/csf-factories#subpath-imports
++ import preview from '#.storybook/preview';
+- import { Meta, StoryObj } from '@storybook/your-renderer';
+
+import { Button } from './Button';
+
++ const meta = preview.meta({
+- const meta = {
+    // ...current meta
++ });
+- } satisfies Meta<typeof Button>;
+- export default meta;
+
+- type Story = StoryObj<typeof meta>;
+
++ export const Primary = meta.story({
+- export const Primary: Story = {
+    // ...current story
++ });
+- };
+```
+
+Note how it is no longer necessary to import any types or manually apply types to the meta or stories. The types are now inferred automatically, thanks to the factory function pattern.
+
+#### 4.1 Reusing story properties
+
+Previously, story properties such as `Story.args` or `Story.parameters` were accessed directly when reusing them in another story. While accessing them like this is still supported, it is deprecated in CSF Factories.
+
+All of the story properties are now contained within a new property called `composed` and should be accessed from that property instead. For instance, `Story.composed.args` or `Story.composed.parameters`.
+
+```diff title="Button.stories.ts"
+// ...rest of file
+
++ export const Primary = meta.story({
+- export const Primary: Story = {
+    args: { primary: true },
++ });
+- };
+
++ export const PrimaryDisabled = meta.story({
+- export const PrimaryDisabled: Story = {
+    args: {
++     ...Primary.composed.args,
+-     ...Primary.args,
+      disabled: true,
+    }
++ });
+- };
+```
+
+<Callout variant="info">
+  The property name "composed" was chosen because the values within are composed from the story, its component meta, and the preview configuration.
```

Comment:
btw the codemod will also change story files like this and will use `.input` instead of `.composed`. I'm assuming that users who originally wrote `MyStory.args` are expecting to see the args that they wrote in the story, not the composed ones coming from preview/meta as well. Though I'd say it's more valuable to get the composed ones!